packages/components/src/spectrum/CheckboxGroup.tsx

```diff
@@ -0,0 +1,39 @@
+import {
+  Checkbox,
+  CheckboxGroup as SpectrumCheckboxGroup,
+  SpectrumCheckboxGroupProps,
+} from '@adobe/react-spectrum';
+import React, { useMemo } from 'react';
+
+/**
+ * Augmented version of the Spectrum CheckboxGroup component that supports
+ * primitive item children.
+ */
+export function CheckboxGroup({
+  children,
+  ...props
+}: SpectrumCheckboxGroupProps): JSX.Element {
+  const wrappedChildren = useMemo(
+    () =>
+      React.Children.map(children, child => {
+        if (typeof child === 'function') {
+          return child;
+        }
+        return <Checkbox>{child}</Checkbox>;
+      }),
+    [children]
+  );
+
+  return (
+    <SpectrumCheckboxGroup
+      // eslint-disable-next-line react/jsx-props-no-spreading
+      {...props}
+      // eslint-disable-next-line react/no-children-prop
+      children={wrappedChildren}
```

Comment:
You should be able to get rid of the eslint exclusions and the children prop and do

```typescript
<SpectrumCheckboxGroup {...props}>
{wrappedChildren}
</SpectrumCheckboxGroup>